components/batch-system/src/metrics.rs

```diff
@@ -11,6 +40,55 @@ lazy_static! {
     )
     .unwrap();
 
+    pub static ref FSM_RESCHEDULE_COUNTER_VEC: IntCounterVec = register_int_counter_vec!(
+        "tikv_batch_system_fsm_reschedule_total",
+        "Total number of fsm reschedule.",
+        &["type"]
+    )
+    .unwrap();
+    pub static ref FSM_RESCHEDULE_COUNTER: FsmRescheduleCounterVec =
+        auto_flush_from!(FSM_RESCHEDULE_COUNTER_VEC, FsmRescheduleCounterVec);
+
+    pub static ref FSM_SCHEDULE_WAIT_DURATION_VEC: HistogramVec =
+    register_histogram_vec!(
+        "tikv_batch_system_fsm_schedule_wait_seconds",
+        "Duration of fsm waiting to schedule.",
+        &["type"],
+        exponential_buckets(0.001, 1.59, 20).unwrap(), // max 10s
+    ).unwrap();
+    pub static ref FSM_SCHEDULE_WAIT_DURATION: FsmScheduleWaitDurationVec =
+    auto_flush_from!(FSM_SCHEDULE_WAIT_DURATION_VEC, FsmScheduleWaitDurationVec);
+
+    pub static ref FSM_POLL_DURATION_VEC: HistogramVec =
+    register_histogram_vec!(
+        "tikv_batch_system_fsm_poll_seconds",
+        "Duration of one fsm being polled.",
```

Comment:
This is probably more descriptive and easier to understand?
```suggestion
        "Total time for an FSM to finish processing all messages, potentially over multiple polling rounds.",
```